spring-integration-xmpp/src/main/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpoint.java

```diff
@@ -64,11 +70,37 @@ public void setHeaderMapper(XmppHeaderMapper headerMapper) {
 	 * Specify whether the text message body should be extracted when mapping to a
 	 * Spring Integration Message payload. Otherwise, the full XMPP Message will be
 	 * passed within the payload. This value is <em>true</em> by default.
-	 *
 	 * @param extractPayload true if the payload should be extracted.
+	 * @deprecated since version 4.3 in favor of {@link #setPayloadExpression(Expression)}
 	 */
+	@Deprecated
 	public void setExtractPayload(boolean extractPayload) {
-		this.extractPayload = extractPayload;
+		if (this.payloadExpression == null) {
+			setPayloadExpression(extractPayload ?
+					EXPRESSION_PARSER.parseExpression("payload")
+					: EXPRESSION_PARSER.parseExpression("#this"));
+		}
```

Comment:
He-he! I was thought about some strategy (Lambda) instead of `Expression`, but that ends up with a lot of them throughout the project. The Java 8 with `Fuction` may make our life easier...

So, I've decided to leave just general solution with the `Expression` just because there is a `FunctionExpression` in DSL, when you just can place it into the classpath and use as is.

It will be here eventually anyway, but I don't see any problems to use one more dependency if you'd like to get a gain from the `FunctionExpression`.